migrate/20240523_vm_display_state_waiting_for_capacity.rb

```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+Sequel.migration do
+  change do
+    add_enum_value(:vm_display_state, "waiting for capacity")
```

Comment:
I have a plan to remove `display_state` column and make it dynamic similar to the `display_state` of `PostgresResource` https://github.com/ubicloud/ubicloud/blob/82fc2280d2411bc9af9d0a95904e88eabcb0a0f7/model/postgres/postgres_resource.rb#L45.

Is there an easy way to add this state here: https://github.com/ubicloud/ubicloud/blob/82fc2280d2411bc9af9d0a95904e88eabcb0a0f7/model/vm.rb#L54

Reply:
that should be possible. one way would be to introduce a semaphore that is incremented when we cannot allocate a host for the vm, and decremented again after successful allocation.

Reply:
I don't think it's necessary for this PR. It's good to see it's doable. We can do it when we remove the `display_state` column.